### PR TITLE
[Connector-V2] Add parquet writer in file connector

### LINKED ISSUE #2196: [Connector-V2] Add parquet FileFormat for File Sink Connector
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

#1946 

Now the FileSink Connector only support text file format, `PARQUET` is a very common file format and it often used in hive. We need support `PARQUET` file format.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

#2196 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by EricJoy2048 — COMMENTED** [review]
+1

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalParquetTransactionStateFileWriter.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            } else {
+                recordBuilder.set(seaTunnelRowTypeInfo.getFieldName(index), seaTunnelRow.getField(index));
+            }
+        });
+        GenericData.Record record = recordBuilder.build();
+        try {
+            writer.write(record);
+        } catch (IOException e) {
+            LOGGER.error("Write data to file [{}] error", filePath);
+            throw new RuntimeException(e);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalParquetTransactionStateFileWriter.java]: It's better add Exception msg.

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalTransactionStateFileWriteFactory.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    partitionDirNameGenerator,
+                    sinkColumnsIndexInRow,
+                    tmpPath,
+                    targetPath,
+                    jobId,
+                    subTaskIndex,
+                    fieldDelimiter,
+                    rowDelimiter,
+                    fileSystem);
+        } else if (fileFormat.equals(FileFormat.PARQUET)) {
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalTransactionStateFileWriteFactory.java]: meaningless `else` code

File: seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/hdfs/HdfsParquetTransactionStateFileWriter.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            } else {
+                recordBuilder.set(seaTunnelRowTypeInfo.getFieldName(index), seaTunnelRow.getField(index));
+            }
+        });
+        GenericData.Record record = recordBuilder.build();
+        try {
+            writer.write(record);
+        } catch (IOException e) {
+            LOGGER.error("Write data to file [{}] error", filePath);
+            throw new RuntimeException(e);
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/hdfs/HdfsParquetTransactionStateFileWriter.java]: If an exception needs to be thrown, then there is no need to use Log.error

**Review by EricJoy2048 — CHANGES_REQUESTED** [review]
In addition, you need to modify the document of File Sink and add the information about parquet support.

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalTransactionStateFileWriteFactory.java (hunk 2 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalTransactionStateFileWriteFactory.java]: done

File: seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/hdfs/HdfsParquetTransactionStateFileWriter.java (hunk 3 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/hdfs/HdfsParquetTransactionStateFileWriter.java]: done

**Comment by TyrantLucifer** [comment]
> In addition, you need to modify the document of File Sink and add the information about parquet support.

done

File: seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalParquetTransactionStateFileWriter.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-local/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/local/LocalParquetTransactionStateFileWriter.java]: done

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**